src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py

```diff
@@ -144,3 +146,43 @@ def Create(self, cmd, client, resource_group_name: str, cluster_name: str, name:
             location=""
         )
         return extension_instance, release_name, create_identity
+
+    def Update(self, cmd, resource_group_name: str, cluster_name: str, auto_upgrade_minor_version: bool,
+               release_train: str, version: str, configuration_settings: dict,
+               configuration_protected_settings: dict, original_extension: Extension, yes: bool = False) \
+            -> PatchExtension:
+        """ExtensionType 'Microsoft.Dapr' specific validations & defaults for Update.
+           Must create and return a valid 'PatchExtension' object.
+        """
+        input_configuration_settings = deepcopy(configuration_settings)
+
+        # configuration_settings can be None, so we need to set it to an empty dict.
+        if configuration_settings is None:
+            configuration_settings = {}
+
+        # If we are downgrading the extension, then we need to disable the apply-CRDs hook.
+        # This is because CRD updates while downgrading can cause issues.
+        # As CRDs are additive, skipping their removal while downgrading is safe.
+        original_version = original_extension.version
+        if self.APPLY_CRDS_HOOK_ENABLED_KEY in configuration_settings:
```

Comment:
This means if we are downgrading then also we can apply crds.
Should we do this or - 
- in case of downgrade always skip the crds and ignore what has been passed.
- and in case of upgrade take what has been passed and if not passed, then set it to true. WDYT ?

Reply:
Ideally users should not pass this configuration explicitly and let the CLI figure it out. 

However in case they do pass it, I would propose having a consistent behavior. This can also be useful when a given downgrade path is compatible with CRD downgrade. Even if we block it, users can always use `Kubectl` to do it.

What do you think?

Reply:
ok.. I agree with maintaining consistent behaviour.